itests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/adapter/SpringBootTest.java

```diff
@@ -0,0 +1,217 @@
+import org.jboss.tools.openshift.reddeer.requirement.CleanOpenShiftExplorerRequirement.CleanOpenShiftExplorer;
+import org.jboss.tools.openshift.reddeer.requirement.OpenShiftCommandLineToolsRequirement.OCBinary;
+import org.jboss.tools.openshift.reddeer.requirement.OpenShiftConnectionRequirement;
+import org.jboss.tools.openshift.reddeer.requirement.OpenShiftConnectionRequirement.RequiredBasicConnection;
+import org.jboss.tools.openshift.reddeer.requirement.OpenShiftProjectRequirement;
+import org.jboss.tools.openshift.reddeer.requirement.OpenShiftProjectRequirement.RequiredProject;
+import org.jboss.tools.openshift.reddeer.utils.OpenShiftLabel;
+import org.jboss.tools.openshift.reddeer.view.OpenShiftExplorerView;
+import org.jboss.tools.openshift.reddeer.view.resources.OpenShiftProject;
+import org.jboss.tools.openshift.reddeer.view.resources.OpenShiftResource;
+import org.jboss.tools.openshift.reddeer.view.resources.ServerAdapter;
+import org.jboss.tools.openshift.reddeer.view.resources.ServerAdapter.Version;
+import org.jboss.tools.openshift.ui.bot.test.application.v3.basic.AbstractTest;
+import org.jboss.tools.openshift.ui.bot.test.common.OpenShiftUtils;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@OpenPerspective(value=JBossPerspective.class)
+@RunWith(RedDeerSuite.class)
+@OCBinary(cleanup=false, setOCInPrefs=true)
+@CleanOpenShiftExplorer
+@RequiredBasicConnection
+@CleanConnection
+@RequiredProject(name="fuse-on-openshift")
+
+/*
+ * See: https://issues.jboss.org/browse/JBIDE-25303
+ * 
+ * @author jkopriva@redhat.com
+ */
+public class SpringBootTest extends AbstractTest  {
+
+	public static String PUBLISHED_CODE = "package org.mycompany;\n" + 
+			"\n" + 
+			"import org.springframework.stereotype.Component;\n" + 
+			"\n" + 
+			"/**\n" + 
+			" * A sample transform\n" + 
+			" */\n" + 
+			"@Component(value = \"myTransformer\")\n" + 
+			"public class MyTransformer {\n" + 
+			"\n" + 
+			"    public String transform() {	\n" + 
+			"    	return \"NewCode\";\n" + 
+			"    }\n" + 
+			"\n" + 
+			"}";
+	
+	private static final String PROJECT_NAME = "fuse-on-openshift";
+	private static final String FUSE_SERVICE_NAME = PROJECT_NAME;
+	
+	@InjectRequirement
+	private static OpenShiftConnectionRequirement connectionReq;
+	
+	@InjectRequirement
+	private static OpenShiftProjectRequirement projectReq;
+	
+	@BeforeClass	
+	public static void waitForRunningApplication() {
+		new ProjectExplorer().deleteAllProjects(true);
+	}
+
+	@Test
+	public void testSpringBootApplication() {
+		importProjectFromZip();
+		editAndRunLaunchConfiguration();
+		podIsRunningAndCheckLogChanging();
+		createServerAdapterAndWaitForStart();
+		podIsRunningAndCheckLog("OldCode");
+		changeSourceCode();
+		podIsRunningAndCheckLog("NewCode");
+	}
+	
+	private void importProjectFromZip() {
```

Comment:
Later, it might be moved to some utils or parent class when other test cases will require to use this functionality. But in future of course...